File: operation-nest/entity/expected-parcel-operation/expected-parcel-operation.ts
import { IsBase64, IsInt, IsNotEmpty, Max, Min } from 'class-validator';
import { IsEncryptedNumber } from '../../../cryptography/is-encrypted-number';
import { OperationDto } from '../../core/domain/operation.dto';

export class ExpectedParcelOperation extends OperationDto {

    @IsNotEmpty()
    @IsInt()
    @Min(1900)
    @Max(2100)
    year: number;

    @IsNotEmpty()
    @IsInt()
    @Min(1)
    @Max(12)
    month: number;

    @IsNotEmpty()
    @IsBase64()
    @IsEncryptedNumber()
    monthlySum: string;

    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
    constructor(data?: any) {
        super(data);
        this.year = data && data.year || null;
        this.month = data && data.month || null;
        this.monthlySum = data && data.montlhySum || null;
    }
}

File: operation-spring/entity/dto/ExpectedParcelOperation.java
package com.prompt.operation.entity.dto;

import com.prompt.operation.core.dto.OperationDto;

import java.time.LocalDateTime;

public class ExpectedParcelOperation extends OperationDto {

    private Long month;
    private Long year;
    private byte[] monthlySum;

    public ExpectedParcelOperation(LocalDateTime occurredAt, Long month, Long year, byte[] monthlySum) {
        super(occurredAt);
        this.month = month;
        this.year = year;
        this.monthlySum = monthlySum;
    }

    public ExpectedParcelOperation() {
    }

    public Long getMonth() {
        return month;
    }

    public void setMonth(Long month) {
        this.month = month;
    }

    public Long getYear() {
        return year;
    }

    public void setYear(Long year) {
        this.year = year;
    }

    public byte[] getMonthlySum() {
        return monthlySum;
    }

    public void setMonthlySum(byte[] monthlySum) {
        this.monthlySum = monthlySum;
    }
}
